lib/pantry/handlers/list_intent_request_handler.rb

```diff
@@ -0,0 +1,97 @@
+module Pantry
+
+  module Handlers
+
+    class ListIntentRequestHandler < RequestHandler
+
+      def process
+        get_intent_specific_method(intent['name']).call
+      end
+
+      private
+
+      def get_intent_specific_method(intent_name)
+        {
+          'Warning'       => method(:list_warning_items).to_proc,
+          'Expired'       => method(:list_expired_items).to_proc,
+          'FromCategory'  => method(:list_items_from_category).to_proc,
+          'AllItems'      => method(:list_all_items).to_proc
+        }[intent_name]
```

Comment:
You can avoid the conversion to a proc by just mapping the symbol of the method name, then in your `#process` method, doing:
```ruby
send(get_intent_specific_method(intent['name']))
```